src/lambda/local/configureLocalLambda.ts

```diff
@@ -0,0 +1,245 @@
+import { accessAsync, mkdirAsync, writeFileAsync } from '../../shared/filesystem'
+import { DefaultSettingsConfiguration } from '../../shared/settingsConfiguration'
+
+export interface HandlerConfig {
+    event: any
+}
+
+export interface HandlersConfig {
+    handlers: {
+        [ handler: string ]: HandlerConfig | undefined
+    }
+}
+
+export async function configureLocalLambda(sourceUri: vscode.Uri, handler: string): Promise<void> {
+    // Handler will be the fully-qualified name, so we also allow '.' despite it being forbidden in handler names.
+    if (/[^\w\-\.]/.test(handler)) {
+        throw new Error(
+            `Invalid handler name: '${handler}'. ` +
+            'Handler names can contain only letters, numbers, hyphens, and underscores.'
+        )
+    }
+
+    const workspaceFolder: vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(sourceUri)
+    if (!workspaceFolder) {
+        console.error(`Source file ${sourceUri} is external to the current workspace.`)
+
+        return
+    }
+
+    // Preserve the scheme, etc of the workspace uri.
+    const uri = workspaceFolder.uri.with({
+        path: path.join(workspaceFolder.uri.fsPath, '.aws', 'handlers.json')
+    })
+
+    try {
+        await accessAsync(path.dirname(uri.fsPath))
+    } catch {
+        await mkdirAsync(path.dirname(uri.fsPath), { recursive: true })
+    }
+
+    try {
+        await accessAsync(uri.fsPath)
+    } catch {
+        await writeFileAsync(uri.fsPath, JSON.stringify(buildHandlersConfig(handler), undefined, getTabSize()))
+    }
+
+    const editor: vscode.TextEditor = await vscode.window.showTextDocument(uri)
+    if (await prepareConfig(editor, handler)) {
+        // Perf: TextDocument.save is smart enough to no-op if the document is not dirty.
+        await editor.document.save()
+    } else {
+        throw new Error(`Could not update config file '${uri.fsPath}'`)
+    }
+
+    await vscode.window.showTextDocument(
+        editor.document,
+        { selection: await getFocusRange(editor, handler) }
+    )
+}
+
+function buildHandlersConfig(handler?: string): HandlersConfig {
+    const config: HandlersConfig = {
+        handlers: {}
+    }
+
+    if (!!handler) {
+        config.handlers[handler] = buildHandlerConfig()
+    }
+
+    return config
+}
+
+function buildHandlerConfig(): HandlerConfig {
+    return {
+        event: {}
+    }
+}
+
+function getTabSize(tabSize?: string | number | undefined): number {
+    switch (typeof tabSize) {
+        case 'number':
+            return tabSize
+        case 'string':
+            return Number.parseInt(tabSize, 10)
+        default:
+            // If we couldn't determine the tabSize at the document, workspace, or user level, default to 4.
+            return new DefaultSettingsConfiguration('editor').readSetting<number>('tabSize') || 4
+    }
+}
+
+async function prepareConfig(editor: vscode.TextEditor, handler: string): Promise<boolean> {
```

Comment:
This is hard to digest. I don't know if a little more commentary would help, or perhaps if it was placed into a class of its own, so that the different cases can be parceled out to separate methods.

Reply:
✔️ 

I split out some parts of this function, simplified the rest and added more commentary. Let me know if you still find it hard to digest!